### ISSUE TITLE
Avoid getting default configuration using command line builds iOS

### ISSUE THREAD

**Issue by zaidas1m** [flutter/flutter#83704]
Hello!

I need to make several deploys with different bundle id for iOS and I have several configurations and schemes set with different bundle id:

![Captura de pantalla 2021-06-01 a las 12 09 06](https://user-images.githubusercontent.com/65656518/120307395-39e30780-c2d3-11eb-8ab3-3b87f42fac79.png)

![Captura de pantalla 2021-06-01 a las 12 12 41](https://user-images.githubusercontent.com/65656518/120307241-0902d280-c2d3-11eb-94ad-365ef06eabc9.png)

but I can't run `flutter build ios` with a specific _flavor_ because always is going to get the configuration (and its bundle) set in "Use for command-line builds" and that makes it unusable.

Is there a way to build ios using command line with flutter like `flutter build ios --flavor <my-other-flavor> ` avoiding getting default option set in this select?
![Captura de pantalla 2021-06-01 a las 12 11 02](https://user-images.githubusercontent.com/65656518/120306742-72ceac80-c2d2-11eb-8d48-f995b157d059.png)
and getting the configuration set for the flavor used in the command line?

Thanks!

**Comment by TahaTesser** [comment]
Hi @zaidas1m 
Please check [this](https://medium.com/@animeshjain/build-flavors-in-flutter-android-and-ios-with-different-firebase-projects-per-flavor-27c5c5dac10b), you can get non-default config  `flutter build ios --flavor <your_favor`


Are you saying if you build `flutter build ios --flavor release-dev`, you always get `release-test` instead?

**Comment by zaidas1m** [comment]
Yes, that is what is happening.

_Release-Dev_ configuration has bundle id = com.swiss1mobile.cablex-toolbox
_Release-Test_ configuration has bundle id = com.swiss1mobile.cablex-toolbox-test
But always gets what is set in that selector "Use for command-line builds".

and the console...
I have set _Release-Test_ and whatever flavor I use, it builds bundle id = _com.swiss1mobile.cablex-toolbox-test_: 

![Captura de pantalla 2021-06-01 a las 14 46 13](https://user-images.githubusercontent.com/65656518/120326361-ef6c8580-c2e8-11eb-966e-91058ba09be8.png)

Now if I change in that selector and I set _Release-Dev_, whatever flavor I use it builds _com.swiss1mobile.cablex-toolbox_:

![Captura de pantalla 2021-06-01 a las 14 48 39](https://user-images.githubusercontent.com/65656518/120326538-22167e00-c2e9-11eb-90da-d6d03ebd1ef8.png)

Bundle id is different for Dev and for Test and I need to get the specific one when is making the build, otherwise using flavor doesn't work at all for `flutter build` command.

Thanks for replying and for the link. I have already all schemes and product bundle identifiers configured.
Not sure if could a problem of naming. I don't know...

**Comment by TahaTesser** [comment]
Hi @zaidas1m 
Can you please provide a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) in a repository
Thank you

**Comment by droidpl** [comment]
I am getting the same issue. No matter which flavor I set, the build or ipa commands are always returning the default configuration set.

**Comment by mvrsn** [comment]
I'm experiencing this issue as well. I'm not sure when this stopped working as I didn't build an IPA for my project for some time, but now the `flavor` argument seems to be completely ignored, and the only way to build the correct flavor is to change the setting as shown in the screenshot by the OP

Some recent changes that I've done that could have impacted my project is to enable sound null-safety and updating Flutter to the latest version, and I also recently renamed my app including changing the namespace/package 

Also came across https://stackoverflow.com/questions/67512656/build-flavors-always-defaults-to-command-line-default-on-ios-flutter, so it seems like there are at least a few people that are having problems with this

**Comment by mvrsn** [comment]
I found [this article](https://medium.com/flutter-community/flavoring-flutter-applications-android-ios-ea39d3155346) which links to [this repository](https://github.com/ptyagicodecamp/counterapp_flavors) where the problem is (at least in my environment) reproducible

By default the `Debug-prod` flavor is set when opening the project in XCode, and when running `flutter build ipa -t lib/main_dev.dart --flavor dev` the build output says `Archiving com.pcc.counterapp.prod...`

**Comment by ikurek** [comment]
It looks like `--flavor` param isn't passed to `xcodebuild` as `-schema`, so default value set in XCode is used. The default value is in `Runner.xcodeproj/project.pbxproj` under the `defaultConfigurationName`, so ugly solution for those desperate to fix this:
1. Open `project.pbxproj` as text file
2. Replace all `defaultConfigurationName` values with build schema that should be used
3. Save `project.pbxproj`
4. Run `flutter build ios --flavor <flavor name>`
5. Build runs with default schema defined in step 2

This is a nasty hack, but in my case, the issue makes whole CI/CD useless, so finding any solution was good. This can also be easily done automatically using a simple script.

Please fix this ASAP, being unable to build the app properly is very annoying.

**Comment by jmagman** [comment]
> It looks like `--flavor` param isn't passed to `xcodebuild` as `-schema`, so default value set in XCode is used. 

https://github.com/flutter/flutter/pull/80904 is available on 2.3.0-1.0.pre, which is currently available on the [beta channel](https://flutter.dev/docs/development/tools/sdk/releases?tab=macos).

I see @TahaTesser reproduced on master, I will take a look with the reproduction project.

**Comment by TahaTesser** [comment]
Hi @zaidas1m @ikurek @mvrsn
 I have a created fresh project following this [article](https://medium.com/flutter-community/flavoring-flutter-applications-android-ios-ea39d3155346) and tested the schemes with `flutter build ios  --flavor dev -t lib/main_dev.dart`
 
 In xcode, seleect `Release-Prod` for "Use xxx for command line tools
 
 When running the build on latest stable `2.2.3`  for dev scheme, I see prod scheme is building 
 ```console
tahatesser@Tahas-MBP triage-companion % fs build ios  --flavor dev -t lib/main_dev.dart
Running "flutter pub get" in triage-companion...                 1,103ms
Building com.tahatesser.triage.prod for device (ios-release)...
Automatically signing iOS for device deployment using specified development team
```
However https://github.com/flutter/flutter/issues/83704#issuecomment-883732497 mentioned, I see correct scheme is building on `2.3` (current latest beta)
```console
tahatesser@Tahas-MBP triage-companion % fb build ios  --flavor dev -t lib/main_dev.dart
Running "flutter pub get" in triage-companion...                 2,512ms
Building com.tahatesser.triage.dev for device (ios-release)...
Automatically signing iOS for device deployment using specified development team
```

<details> 
<summary>flutter doctor -v</summary> 
	
```console
[✓] Flutter (Channel beta, 2.3.0-24.1.pre, on macOS 11.4 20F71 darwin-x64,
    locale en-US)
    • Flutter version 2.3.0-24.1.pre at /Users/tahatesser/Code/flutter_beta
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 615957513e (4 weeks ago), 2021-06-25 09:24:44 -0700
    • Engine revision 9d517f475b
    • Dart version 2.14.0 (build 2.14.0-188.5.beta)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Volumes/Extreme/android-sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_SDK_ROOT = /Volumes/Extreme/android-sdk
    • Java binary at: /Users/tahatesser/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.
[✓] Xcode - develop for iOS and macOS
    • Xcode at /Volumes/Extreme/Xcode.app/Contents/Developer
    • Xcode 12.5, Build version 12E262
    • CocoaPods version 1.10.1
[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
[✓] Android Studio (version 4.2)
    • Android Studio at /Users/tahatesser/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
[✓] Android Studio (version 4.2)
    • Android Studio at /Users/tahatesser/Library/Application
      Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
[✓] VS Code (version 1.58.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.24.0
[✓] Connected device (3 available)
    • Taha's iPhone (mobile) • 00008020-001059882212002E • ios            • iOS
      14.6
    • macOS (desktop)        • macos                     • darwin-x64     •
      macOS 11.4 20F71 darwin-x64
    • Chrome (web)           • chrome                    • web-javascript •
      Google Chrome 91.0.4472.164
• No issues found!
```

</details>

Can you please upgrade to the latest  `beta` and try again
Execute the following commands to upgrade
`flutter channel beta`
`flutter upgrade --force`
`flutter doctor -v`



In my earlier testing, I was using old project that migh have had some configuration issues but I can confirm with fresh setup, this is fixed. If the problem persist, ping me, I can provide a public repo example 
Closing as fixed,  If you disagree, please write in the comments and I will reopen it.
Thank you
